src/main/java/formflow/library/ScreenController.java

```diff
@@ -683,6 +748,359 @@ ModelAndView deleteSubflowIteration(
+              subflowIterationUuid, currentScreen.getSubflow(), submission.getFlow(), currentScreen.getName()));
+    }
+
+    if (!isValidRepeatForIterationUuid(submission, flowName,
+        currentScreen.getSubflow(),
+        subflowIterationUuid, repeatFor.getSaveDataAs(), repeatForIterationUuid)) {
+      throwNotFoundError(submission.getFlow(), currentScreen.getName(),
+          String.format("repeatFor iteration uuid ('%s') is not valid for subflow '%s' iteration ('%s') in flow %s)",
+              repeatForIterationUuid, currentScreen.getSubflow(), subflowIterationUuid, flowName));
+    }
+
+    if (shouldRedirectDueToLockedSubmission(screenName, submission, flow)) {
+      String lockedSubmissionRedirectUrl = getLockedSubmissionRedirectUrl(flow, redirectAttributes, locale);
+      return new ModelAndView("redirect:" + lockedSubmissionRedirectUrl);
+    }
+
+    String nextViewableScreen = getNextViewableScreen(flow, screenName, subflowIterationUuid, repeatForIterationUuid,
+        submission);
+    if (!nextViewableScreen.equals(screenName)) {
+      return new ModelAndView(
+          String.format("redirect:/flow/%s/%s/%s/%s", flow, nextViewableScreen, subflowIterationUuid,
+              repeatForIterationUuid));
+    }
+
+    if (submission == null) {
+      // we have issues! We should not get here, really.
+      log.error("There is no submission associated with request!");
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+    }
+
+    actionManager.handleBeforeDisplayAction(currentScreen, submission, validatedSubflowIterationUuid,
+        repeatForIterationUuid);
+    Map<String, Object> model = createModel(flow, screenName, httpSession, submission, validatedSubflowIterationUuid,
+        repeatForIterationUuid, request,
+        referer);
+    model.put("formAction",
+        String.format("/flow/%s/%s/%s/%s", flow, screenName, validatedSubflowIterationUuid,
+            repeatForIterationUuid));
+    return new ModelAndView(String.format("%s/%s", flow, screenName), model);
+  }
+
+  private void validateSubflowRelationshipConfiguration(Optional<SubflowRelationship> subflowRelationshipOptional,
+      String subflowName) {
+    if (subflowRelationshipOptional.isEmpty()) {
+      throw new IllegalStateException(String.format(
+          "Configuration Error: Missing Relationship definition in configuration file for subflow %s",
+          subflowName));
+    }
+
+    final SubflowRelationship subflowRelationship = subflowRelationshipOptional.get();
+
+    if (subflowRelationshipOptional.isPresent() && subflowRelationship.getRepeatFor() == null) {
+      throw new IllegalStateException(String.format(
+          "Configuration Error: Missing repeatFor definition for the nested relationship in configuration file for subflow %s",
+          subflowName));
+    }
+  }
+
+  /**
+   * Processes input data from a page of a subflow screen. If `new` is supplied for UUID, then it is assumed this is a new
```

Comment:
Does `new` ever get passed here since we are pre-populating the data in the /flow/screen/outerUuid endpoint?